operator/cmd/root.go

```diff
@@ -405,16 +412,14 @@ func OnOperatorStartLeading(ctx context.Context, clientset k8sClient.Clientset)
 	}
 
 	if kvstoreEnabled() {
-		if operatorOption.Config.SyncK8sServices {
-			operatorWatchers.StartSynchronizingServices(clientset, true, option.Config)
-		}
-
 		var goopts *kvstore.ExtraOptions
 		scopedLog := log.WithFields(logrus.Fields{
 			"kvstore": option.Config.KVStore,
 			"address": option.Config.KVStoreOpt[fmt.Sprintf("%s.address", option.Config.KVStore)],
 		})
-		if operatorOption.Config.SyncK8sServices {
+
+		if k8s.IsEnabled() && operatorOption.Config.SyncK8sServices {
```

Comment:
Maybe another question, I see `SyncK8sServices` as a config option here. What's the difference between this and the new option?

Reply:
Ah I see, it makes sense to me for there to be one toggle for all k8s logic. When I read `enable-k8s-services`, I assumed that this meant only the "k8s service" (virtual IP / load balancing / SNAT / DNAT) type logic. Reading your response above, it sounds like you meant "services" in the sense that these are features that k8s provides. Could we just use a new flag something like `--enable-k8s` instead? Then it would be clear that it is indicating _all_ kubernetes support and remove the name conflict with a kubernetes-specific feature (Services).

Otherwise :+1: makes sense. We are currently working on refactoring the logic into more modular and convenient "Cells" (basically just application modules with self-contained logic). This effort should make this code simpler over time, but I don't think that your PR needs to wait for such an effort. Something close to what you have proposed already seems fine to me. I will CC @joamaki just as a heads up that the ability to disable k8s entirely is a community ask, in case it impacts that refactoring effort.

Reply:
I vote for `--disable-k8s` (false by default)

Reply:
We are trying to avoid double-negatives in the codebase, because they are frequently confusing to developers and users. This is why I'm suggesting `--enable-k8s` with a default to true.

Reply:
I'll update tomorrow with --enable-k8s (w/ default=true) :)